### ISSUE TITLE
Detect Nested percent literals

### ISSUE THREAD

**Issue by dwilkins** [bbatsov/rubocop#4628]
Detect Nested percent literals
--------
Nesting Percent Literals silently does the wrong thing.   It would be nice for Rubocop to catch this error
## Expected behavior

Emit an offense when a percent literal is used inside of another percent literal

## Actual behavior

No offense is emitted

## Steps to reproduce the problem
```ruby
#! /bin/env ruby
# test_percent_literals.rb
require 'json'

attributes = [
  valid_attributes: %i[name content],
  nested_attributes: %i[name department
    incorrectly_nested_attributes %i[
    name organization etc]]
]

puts JSON.pretty_generate(attributes)
``` 
`rubocop test_percent_literals.rb`

## RuboCop version
```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by asherkach** [comment]
This seems straightforward, I'll build out a new cop for this. Do you see this as emitting a rubocop offense for all nestings of all [percent literals](https://ruby-doc.org/core-2.1.1/doc/syntax/literals_rdoc.html#label-Percent+Strings)? In particular:
1. Should an offense be emitted only if the outer and inner percent character modifiers are the same? E.g., should `%w[a %i[x y] b]` emit an offense?
1. Should an offense be emitted only for certain pairs of outer percent character modifiers and inner percent character modifiers? If so, which ones (not all combinations seem equally error prone)? E.g., should `%x[a %i(x y) b]` emit an offense?
1. Should an offense be raised only if the percent literal is followed by a delimiter? E.g., should [`%w[% %i %I %q %Q %r %s %w %W %x]`](https://github.com/bbatsov/rubocop/blob/022e7322b731b06bd31ce5ef1bac378b27551ed0/lib/rubocop/cop/mixin/percent_literal.rb#L7) emit an error?

**Comment by Drenmi** [comment]
@asherkach: I recommend building the basic implementation first, that catches all instances. It's very easy to go down the rabbit hole trying to add all configuration options at once, and it ends up in a never-ending PR review.

It is always possible to add more functionality in subsequent PRs. 🙂

**Comment by bquorning** [comment]
I don’t see this as a very common problem. And if it should happen, my guess is that the developer will quickly notice (proper tests would fail) and fix it.
